index.js

```diff
@@ -68,46 +68,39 @@ function generateFiveDayForecast() {
-            document.querySelectorAll("ul > li")[0].appendChild(firstDayNewTemp);
-            firstDayNewTemp.style.marginLeft = "70px";
-            let firstDayIcon = data.list[0].weather[0].icon;
-            let firstImageUrl = "http://openweathermap.org/img/wn/" + firstDayIcon + "@2x.png";
-            let firstNewIcon = document.createElement("span");
-            firstNewIcon.innerHTML = "<img src=" + firstImageUrl + ">";
-            document.querySelectorAll("ul > li")[0].appendChild(firstNewIcon);
-            firstNewIcon.style.marginLeft = "30px";
-            let secondDayNewTemp = document.createElement("span");
-            secondDayNewTemp.innerHTML = data.list[2].main.temp;
-            document.querySelectorAll("ul > li")[1].appendChild(secondDayNewTemp);
-            secondDayNewTemp.style.marginLeft = "70px";
-            let secondDayIcon = data.list[2].weather[0].icon;
-            let secondImageUrl = "http://openweathermap.org/img/wn/" + secondDayIcon + "@2x.png";
-            let secondNewIcon = document.createElement("span");
-            secondNewIcon.innerHTML = "<img src=" + secondImageUrl + ">";
-            document.querySelectorAll("ul > li")[1].appendChild(secondNewIcon);
-            secondNewIcon.style.marginLeft = "30px";
-            let thirdDayNewTemp = document.createElement("span");
-            thirdDayNewTemp.innerHTML = data.list[10].main.temp;
-            document.querySelectorAll("ul > li")[2].appendChild(thirdDayNewTemp);
-            thirdDayNewTemp.style.marginLeft = "70px";
-            let thirdDayIcon = data.list[10].weather[0].icon;
-            let thirdImageUrl = "http://openweathermap.org/img/wn/" + thirdDayIcon + "@2x.png";
-            let thirdNewIcon = document.createElement("span");
-            thirdNewIcon.innerHTML = "<img src=" + thirdImageUrl + ">";
-            document.querySelectorAll("ul > li")[2].appendChild(thirdNewIcon);
-            thirdNewIcon.style.marginLeft = "30px";
-            let fourthDayNewTemp = document.createElement("span");
-            fourthDayNewTemp.innerHTML = data.list[18].main.temp;
-            document.querySelectorAll("ul > li")[3].appendChild(fourthDayNewTemp);
-            fourthDayNewTemp.style.marginLeft = "70px";
-            let fourthDayIcon = data.list[18].weather[0].icon;
-            let fourthImageUrl = "http://openweathermap.org/img/wn/" + fourthDayIcon + "@2x.png";
-            let fourthNewIcon = document.createElement("span");
-            fourthNewIcon.innerHTML = "<img src=" + fourthImageUrl + ">";
-            document.querySelectorAll("ul > li")[3].appendChild(fourthNewIcon);
-            fourthNewIcon.style.marginLeft = "30px";
+            let tempArray = [];
+            tempArray.push(data.list[1].main.temp);
+            tempArray.push(data.list[2].main.temp);
+            tempArray.push(data.list[10].main.temp);
+            tempArray.push(data.list[18].main.temp);
+
+            function displayTemperature() {
```

Comment:
these two function should be defined outside of `generateFiveDayForecast` function. You would call them - that's correct but not create. Because otherwise on each search your code would recreate these function and call right after.

Reply:
i deleted the generateFiveDayForecast function,as i think this function is not relevant here